backend/main.go

```diff
@@ -280,23 +353,30 @@ func (s *Server) handlePutPeriod(w http.ResponseWriter, r *http.Request) {
 	if !ok {
 		return
 	}
-	if !s.ensureTeamExistence(w, r, teamID, true) {
+	team, exists := s.ensureTeamExistence(w, r, teamID, true)
+	if !exists {
 		return
 	}
+	log.Print(team)
 	savedPeriod, ok := s.ensurePeriodExistence(w, r, teamID, periodID, true)
 	if !ok || !s.ensureNoConcurrentMod(w, r, period, savedPeriod) {
 		return
 	}
 	period.LastUpdateUUID = uuid.New().String()
 	ctx, cancel := context.WithTimeout(r.Context(), s.storeTimeout)
 	defer cancel()
-	err := s.store.UpdatePeriod(ctx, teamID, period)
-	if err != nil {
-		log.Printf("Could not update period '%s' for team '%s': error: %s", periodID, teamID, err)
-		http.Error(w, fmt.Sprintf("Could not update period '%s' for team '%s' (see server log)", periodID, teamID), http.StatusInternalServerError)
-		return
+
+	if s.auth.CanActOnTeam(r.Context().Value("user").(models.User), team, auth.ActionWrite) {
```

Comment:
Can we extract the user to a variable here, just to simplify this line slightly?

Same comment in a few other places.

Reply:
👍🏻